cmds/kexec/kexec_linux.go

```diff
@@ -78,13 +79,12 @@ func main() {
 		opts.exec = true
 	}
 
-	cmdline := opts.cmdline
+	newCmdLine := opts.cmdline
 	if opts.reuseCmdline {
-		procCmdline, err := kexec.CurrentKernelCmdline()
-		if err != nil {
-			log.Fatalf("%v", err)
+		newCmdLine = cmdline.FullCmdLine()
+		if len(newCmdLine) == 0 {
```

Comment:
I'm not too convinced by this interface - it assumes that empty string is a cmdline reading error, while there is an `err` field in the struct. What about just using the `CmdLine` object and check `err`? And if the error is nil just use the cmdline string.

In this case I would also make Raw and Err public - it will simplify the code and push for the use of the `CmdLine` object instead of wrappers around the object